keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1767 @@
+`Allocate` functions.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout or rollback can fail in several ways, primarily affecting new or
+restarting pods rather than impacting the resource allocations of already
+running workloads.
+
+*   **Failure Mode: State File Inconsistency** This feature modifies the on-disk
+    state files (e.g., `/var/lib/kubelet/cpu_manager_state`). A Kubelet restart
+    is required for any rollout or rollback, which introduces risks related to
+    state file compatibility.
+
+    *   **During Rollout (Upgrade):** If the new Kubelet version fails to
+        correctly read an old state file, it could crash on startup or
+        misinterpret existing allocations.
+    *   **During Rollback (Downgrade):** An older Kubelet attempting to read a
+        new, pod-aware state file may fail to parse it. This could lead to a
+        Kubelet crash loop or incorrect resource accounting for all `Guaranteed`
+        pods on the node. The established mitigation for such issues is to drain
+        the node, manually delete the state file, and restart the Kubelet.
+
+*   **Impact on Already Running Workloads** Workloads that are already running
+    and stable are **not directly impacted** by the rollout, as their cgroup
+    settings are already configured and will not be changed. The primary risk is
+    associated with the **Kubelet restart** required for the configuration
+    change. The main impact is on pods that are **restarted** during or after
+    the rollout/rollback, as they will be subject to the new (or old) admission
+    logic and could fail to come back up if an issue is present.
+
+###### What specific metrics should inform a rollback?
+
+When the `PodLevelResourceManagers` feature is enabled, operators should closely
+monitor the following metrics. A sustained or significant increase in any of
+these, particularly the new pod-level error metrics, would be a strong indicator
+that a rollback is necessary:
+
+-   **`pod_topology_manager_pinning_errors_total`**: A rise in this metric
+    indicates that the Topology Manager is failing to find suitable NUMA
+    alignments for pods utilizing pod-level resources. This suggests a
+    fundamental issue with the new hint generation or merging logic.
+-   **`pod_cpu_manager_pinning_errors_total`**: An increase here points to
+    failures in the CPU Manager's new partitioning or allocation logic for
+    pod-level resources. This could manifest as pods failing to acquire their
+    requested CPU sets.
+-   **`pod_memory_manager_pinning_errors_total`**: Similarly, a spike in this
+    metric signals problems with the Memory Manager's new partitioning or
+    allocation for pod-level memory, potentially leading to memory allocation
+    failures.
```

Comment:
I think we should rather have a metric which inform that the resource manager are taking resources from the pod pool (vs the node pool like we had up until now).
I'm not sure I'm overthinking, but my proposal would be to add a new `CounterVec` with a dimension: node vs pod. In the current flow each resource manager allocates container resources from the node pool; with pod level resources they will allocate from the pod pools.

Or perhaps we can just track the allocation from pod resources?